src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -123,12 +126,33 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
         }
         // "path" key in options can incur multiple data read unexpectedly.
         val opts = options - "path"
-        Relation(
-          location.rootPaths.map(_.toString),
-          Hdfs(sourceDataProperties),
-          dataSchema.json,
-          fileFormatName,
-          opts)
+
+        // If user provided a globbing pattern, set rootPaths to this pattern. Else, use file index.
+        val rootPaths = opts.get(GLOBBING_PATTERN_KEY) match {
```

Comment:
I think instead of keeping all paths in rootPaths at creation, 
- keep globPattern in option field in createIndex
- use the globPattern to get the current file list in refreshIndex

WDYT?

Reply:
thanks @sezruby for the suggestion, I thought about that option too and it looked quite lucrative.

The only reason I chose this one was to simplify 2 things:
1. the rest of the index lifecycle / index maintenance gets handled automatically without any other change. This means that we only need to change one class.
2. simpler change in code.

Semantically also, it might be more meaningful to keep rootpaths containing "/temp/*/*" in source.relations. E.g. if the user looks at the metadata, they will immediately know what was the intention when creating index.

Please let me know WDYT. Adding @imback82 , @pirz , @rapoth for opinions/suggestions.

Actually now I also think that if we stick with the current implementation, we can completely remove the globbing config from relation.options. (i.e. one less thing to maintain)

Reply:
ok I realized that without this, Spark keeps all paths in `location.rootPaths`. 
Let's keep the glob paths in rootPaths to avoid duplicate paths in IndexLogEntry.
But with this approach, we still need to keep the globbing path config for refresh; rootPaths will be replaced after refresh, without the config.